.github/workflows/run-profiling.yaml

```diff
@@ -39,20 +39,26 @@ jobs:
       profiling-on-sha: ${{ steps.determine-correct-sha.outputs.result }}
       profiling-event-trigger: ${{ steps.set-github-info.outputs.event }}
     steps:
+      - id: determine-if-pr-comment
+        name: Determine if this is a PR comment
+        run: |
+          if ! [ -z "${{ github.event.issue.pull_request }}" ]; then echo "is-pr=true" >> "${GITHUB_OUTPUT}"; else echo "is-pr=false" >> "${GITHUB_OUTPUT}"; fi
```

Comment:
What happens here if `github.event.issue.pull_request` is not defined? Does GitHub interpolate a zero-length string or raise an error? Can see this would give the correct behaviour if the former but if the latter would this not have a similar issue to previous solution as `github.event.issue` and so `github.event.issue.pull_request`  will only be defined when the triggering event was an issue comment I think?

Reply:
It interpolates an empty string - you can check the (recent) logs in https://github.com/willGraham01/benchmark-action-playtest/actions/workflows/fetch_sha.yml.

That workflow produces no errors in any of the cases (trigger by push, PR comment, issue comment) and returns the expected SHA in each case. Sorry - should've probably linked to this in the main text!

Reply:
Thanks for verifying - this looks goods to go then.